docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,299 @@
+
+## Use of Roles for Login
+
+Handle authentication during your tests with [User Roles](../../guides/advanced-guides/authentication.md#user-roles). Roles allow you to wrap login credentials in a reusable object and use it effectively in tests.
+
+Define roles with the [Role](../../reference/test-api/role/constructor.md) constructor and include them in your tests with the [t.useRole](../../reference/test-api/testcontroller/userole.md) method. The following example makes use of two user roles:
+
+```js
+import { Role } from 'testcafe';
+
+const regularUser = Role('http://example.com/login', async t => {
+    await t
+        .typeText('#login', 'TestUser')
+        .typeText('#password', 'testpass')
+        .click('#sign-in');
+});
+
+const admin = Role('http://example.com/login', async t => {
+    await t
+        .typeText('#login', 'Admin')
+        .typeText('#password', 'adminpass')
+        .click('#sign-in');
+
+fixture `My Fixture`;
+
+test('My test', async t => {
+    await t
+        .useRole(role);
+});
+```
+
+See the [Authentication](../../guides/advanced-guides/authentication.md) article for more in-depth information.
+
+## File Structure
+
+Follow these guidelines to keep your test structure manageable and "clean":
+
+* Use a page model to store Selectors and compound actions that are used often across your app. For instance, put all actions that are necessary to perform an action into one reusable function.
+
+* Put all the page model files into one directory. If your application is divisible into logical chunks (components or subsystems), split up the associated page model objects into separate files.
+
+> You can find an example of a page model in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+* Create a `testcaferc.json` file in the root directory of the project. It enables you to fine-tune your tests, which can be useful in CI/CD systems. For more information, read the [Configuration File](../../reference/configuration-file.md) article.
+
+* Define one `fixture` in every test file. While it's technically possible to define multiple, it may lead to confusion as your suite grows.
+
+* A `fixture`s should aggregate tests that relate to each other by meaning or testing scenario.
+
+* TestCafe tests are purely functional. As such, they should not depend on the implementation details and it is best to isolate them from production code. Keep your test files in a separate directory. You can name this directory appropriately (for instance, `tests`).
+
+* In your test folder, group the tests that cover different subsystems of your application. Separate these into subfolders.
+
+* Don't write long tests. Shorter test scenarios are easier to debug and can run concurrently.
+
+* Any reused data (for example, large sets of reference values or forms inputs) is better stored in a directory of its own. Consider a descriptive folder name (for instance, `data`).
```

Comment:
> in a directory of its own

Maybe *in a separate/dedicated directory*? Because "directory of its own" is too fancy for docs, and can **data** have its **own** directory at all?

Reply:
> ... is better stored in a dedicated directory.